debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationTabGroupViewer.java

```diff
@@ -1449,19 +1452,39 @@ protected void handleTabSelected() {
 		if (fDisposingTabs || fInitializingTabs) {
 			return;
 		}
+		int oldCurrentTabIndex = fCurrentTabIndex;
+		fCurrentTabIndex = fTabFolder.getSelectionIndex();
+
 		ILaunchConfigurationTab[] tabs = getTabs();
-		if (fCurrentTabIndex == fTabFolder.getSelectionIndex() || tabs == null || tabs.length == 0 || fCurrentTabIndex > (tabs.length - 1)) {
+		if (oldCurrentTabIndex == fCurrentTabIndex || tabs == null || tabs.length == 0
+				|| oldCurrentTabIndex > (tabs.length - 1)) {
 			return;
 		}
-		if (fCurrentTabIndex != -1) {
-			ILaunchConfigurationTab tab = tabs[fCurrentTabIndex];
-			ILaunchConfigurationWorkingCopy wc = getWorkingCopy();
-			if (wc != null) {
-				tab.deactivated(wc);
-				getActiveTab().activated(wc);
-			}
+
+		deactivateTab(oldCurrentTabIndex);
+
+		activateCurrentlyActiveTab();
```

Comment:
I am not sure whether I completely understand the change. Isn't it just that the old index shall be assigned to `fCurrentTabIndex` even if method execution already stops at the `return` statement? Then, wouldn't it be sufficient to set that value in the if block? I find the existing code more comprehensible than storing the `oldCurrentTabIndex`. Extracting methods is okay though.

Reply:
The old implementation hid the fact that `fTabFolder.getSelectionIndex()` was the actual current index and that `fCurrentTabIndex` was simply a "left over" which contained the old selected index. This was only made evident in line 1464 before, which flipped the meaning of `fCurrentTabIndex` inside the same method. I find the current way better, more comprehensible.

Reply:
I see, makes sense 👍 Then I would only propose to change `oldCurrentTabIndex` to `previouslyActiveTabIndex` or the like for improved comprehensibility.

Reply:
Puh, kind of a mouthful. Let's meet in the middle: `previousTabIndex` (is even "closer" to `fCurrentTabIndex`) 😉